pkg/kubelet/cm/helpers_linux_test.go

```diff
@@ -197,3 +198,65 @@ func TestMilliCPUToQuota(t *testing.T) {
 		}
 	}
 }
+
+func TestMilliCPUToShares(t *testing.T) {
+	testCases := []struct {
+		input  int64
+		shares uint64
+	}{
+		{
+			input:  int64(0),
+			shares: uint64(2),
+		},
+		{
+			input:  int64(1),
+			shares: uint64(2),
+		},
+		{
+			input:  int64(2),
+			shares: uint64(2),
+		},
+		{
+			input:  int64(5),
+			shares: uint64(5),
+		},
+		{
+			input:  int64(10),
+			shares: uint64(10),
+		},
+		{
+			input:  int64(200),
+			shares: uint64(204),
+		},
+		{
+			input:  int64(1000),
+			shares: uint64(1024),
+		},
+	}
+	for _, testCase := range testCases {
+		result := MilliCPUToShares(testCase.input)
+		if result != testCase.shares {
+			t.Errorf("Input %v, expected shares %v, but got result %v", testCase.input, testCase.shares, result)
+		}
+	}
+}
+
+func TestHugePageLimits(t *testing.T) {
+	resourceList := v1.ResourceList{
+		v1.ResourceHugePagesPrefix + "2Mi":   *resource.NewQuantity(50, resource.BinarySI),
+		v1.ResourceHugePagesPrefix + "4Mi":   *resource.NewQuantity(100, resource.BinarySI),
+		v1.ResourceHugePagesPrefix + "100Mi": *resource.NewQuantity(200, resource.BinarySI),
+	}
+
+	expected := map[int64]int64{
+		2097152:   50,
```

Comment:
It would be better to define a int64 `Mi` globally and use `2*Mi`, `4*Mi` etc here.